dremio_client/model/endpoints.py

```diff
@@ -284,6 +366,142 @@ def user(token, base_url, uid=None, name=None, ssl_verify=True):
+
+
+def update_role(token, base_url, rid, json, ssl_verify=True):
+    """
+    Returns the role after updating it
+    :param token: auth token
+    :param base_url: sql query
+    :param rid: role id
+    :param json: json document for role
+    :param ssl_verify: ignore ssl errors if False
+    :return: result object
+    """
+    return _put(base_url + "/api/v3/role/{}".format(rid), token, json, ssl_verify=ssl_verify)
+
+def get_all_roles(token, base_url ,startIndex=None , count=None ,query=None  , ssl_verify=True):
+    """
+    Fetches all the roles
+    :param token: auth token
+    :param base_url: sql query
+    :param startIndex:(optional) Index from which to start getting roles
+    :param count:(optional) maximum number of roles to fetch
+    :param query: (optional) filter roles based on this query
+    :param ssl_verify: ignore ssl errors if False
+    :return: result object
+    """
+    params=dict()
+    if startIndex is not None:
+        params['startIndex']=startIndex
+    if count is not None:
+        params['count']=count
+    if query is not None:
+        params['query']=query
+    end_url=base_url + "/api/v3/role"
+    qry='&'.join("{}={}".format(k,v) for k,v in params.items())
+    if len(qry)>0:
+        end_url=end_url +'?' + qry
+    return _get(end_url, token,  ssl_verify=ssl_verify)
+
+
+def get_members_of_role(token,base_url ,rid ,startIndex=None, count=None ,ssl_verify=True):
+    """
+    Return members of a given role
+    :param token: auth token
+    :param base_url: sql query
+    :param rid: role id
+    :param startIndex: Index starting from which to return members
+    :param count: Maximum number of members to return
+    :param ssl_verify: Ignore ssl errors if False
+    :return: result object
+    """
+    params=dict()
+    if startIndex is not None:
+        params['startIndex']=startIndex
+    if count is not None:
+        params['count']=count
+    end_url=base_url + "/api/v3/role/{}/member".format(rid)
+    qry='&'.join("{}={}".format(k,v) for k,v in params.items())
+    if len(qry)>0 :
+        end_url = end_url +'?' + qry
+    return _get(end_url , token, ssl_verify=ssl_verify)
+
+def get_privileges_of_role(token,base_url ,rid ,startIndex=None, count=None ,ssl_verify=True):
+    """
+    Fetches privileges of a given role
+    :param token: auth token
+    :param base_url: sql query
+    :param rid: role id
+    :param startIndex: index starting from which to fetch privileges
+    :param count: maximum number of privileges to fetch
+    :param ssl_verify: ignore ssl errors if False
+    :return: result object
+    """
+    params = dict()
+    if startIndex is not None:
+        params['startIndex'] = startIndex
+    if count is not None:
+        params['count'] = count
+    qry = '&'.join("{}={}".format(k, v) for k, v in params.items())
+    end_url = base_url + "/api/v3/role/{}/privilege".format(rid)
+    if len(qry) > 0:
```

Comment:
There is a lot of duplication in these methods. May I suggest something like:

``` python
def build_url(**kwargs):
    query = "&".join("{}={}".format(k,v) for k,v in kwargs.items() if v)
    return "?{}".format(url, query) if query else ""

# replace in methods with
end_url = base_url + "/api/v3/role/{}/privilege".format(rid) + build_url(start_index=start_index, count=count)
```

Reply:
code has been simplified (duplication removed) @rymurr please review